app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java

```diff
@@ -0,0 +1,55 @@
+package com.example.bigowlapp.viewModel;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+import com.example.bigowlapp.model.Group;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.repository.AuthRepository;
+import com.example.bigowlapp.repository.GroupRepository;
+import com.example.bigowlapp.repository.UserRepository;
+
+import java.util.List;
+
+public class MonitoringGroupPageViewModel extends ViewModel {
+
+    private final AuthRepository authRepository;
+    private final GroupRepository groupRepository;
+    private final UserRepository userRepository;
+
+    private MutableLiveData<Group> selectedGroup;
+    private MutableLiveData<List<User>> usersInGroup;
+
+    public MonitoringGroupPageViewModel() {
+        authRepository = new AuthRepository();
+        groupRepository = new GroupRepository();
+        userRepository = new UserRepository();
+    }
+
+    public LiveData<Group> getGroup() {
+        if (selectedGroup == null) {
+            selectedGroup = new MutableLiveData<>();
+            loadGroup();
```

Comment:
Really smart of you to put loadGroup into the method. It's less code for the activity classes.

Reply:
I just did it how it was done in Android documentation:
https://developer.android.com/topic/libraries/architecture/viewmodel#implement